### ISSUE TITLE
Add builtin support for uint8/16/32/64 

### ISSUE THREAD

**Issue by pbrubaker** [ispc/ispc#1510]
Currently we have support for `int8, int16, int32` and `int64`, but no support for unsigned versions.  If we want an unsigned version we have to write `unsigned int32` and it would be much nicer to just type `uint32.`

Can we add support for `uint8, uint16, uint32`, and `uint64`?

**Comment by dbabokin** [comment]
I agree that it makes sense for the sake of conciseness and uniformity of type names. But I'm not completely sure that we should bend the language right now, when it's quite stable. I'll keep the issue open. If other people support it, we should go for it.

**Comment by pbrubaker** [comment]
Totally understood.  Just added it for completeness sake.  Right now I'm typedef'ing them, but it would be handy if they were built in.

**Comment by nnevatie** [comment]
I agree 100% with this request. The very first thing I end up doing in any new ISPC code is:

`#include "types.ispc"`

Which essentially consists of:

```
// Types
typedef unsigned int8  uint8;
typedef unsigned int16 uint16;
typedef unsigned int32 uint32;
typedef unsigned int64 uint64;
```

I'm very used to using the width-specified types in C++ (http://www.cplusplus.com/reference/cstdint/) and the syntactical asymmetry around the signed and unsigned types in ISPC feels very awkward.

**Comment by dbabokin** [comment]
Ok, let's go for it.

**Comment by dbabokin** [comment]
Fixed by PR #1549

**Comment by dbabokin** [comment]
PR #1592 adds macro ISPC_UINT_IS_DEFINED, which may be used to detect is the compiler supports uint* types. This should be handy for projects, which would like to maintain compatibility with multiple ISPC versions. I.e. it may be used as follows:
```c
#ifndef ISPC_UINT_IS_DEFINED
typedef unsigned int64 uint64;
#endif
```
